packages/m/mkl/xmake.lua

```diff
@@ -109,42 +115,57 @@ package("mkl")
         package:add("links", package:is_arch("x64", "x86_64") and "mkl_blas95_" .. suffix or "mkl_blas95")
         package:add("links", package:is_arch("x64", "x86_64") and "mkl_lapack95_" .. suffix or "mkl_lapack95")
 
-        if package:has_tool("cc", "gcc", "gxx") then
-            local flags = {"-Wl,--start-group"}
-            table.insert(flags, package:is_arch("x64", "x86_64") and "-lmkl_intel_" .. suffix or "-lmkl_intel")
-            local threading = package:config("threading")
-            if threading == "tbb" then
-                table.insert(flags, "-lmkl_tbb_thread")
-                package:add("deps", "tbb")
-            elseif threading == "seq" then
-                table.insert(flags, "-lmkl_sequential")
-            elseif threading == "openmp" then
-                table.insert(flags, "-lmkl_intel_thread")
-                table.insert(flags, "-lomp")
-            elseif threading == "gomp" then
-                table.insert(flags, "-lmkl_gnu_thread")
-                table.insert(flags, "-lgomp")
-            end
-            table.insert(flags, "-lmkl_core")
-            table.insert(flags, "-Wl,--end-group")
-            package:add("ldflags", table.concat(flags, " "))
-        else
-            package:add("links", package:is_arch("x64", "x86_64") and "mkl_intel_" .. suffix or "mkl_intel_c")
-            local threading = package:config("threading")
-            if threading == "tbb" then
-                package:add("links", "mkl_tbb_thread")
-                package:add("deps", "tbb")
-            elseif threading == "seq" then
-                package:add("links", "mkl_sequential")
-            elseif threading == "openmp" then
-                package:add("links", "mkl_intel_thread", "omp")
-            elseif threading == "gomp" then
-                package:add("links", "mkl_gnu_thread", "gomp")
+        for _, toolkind in ipairs({"ld", "fcld"}) do
+            if package:config("runtime") == "default" then
+                if (package:has_tool(toolkind, "gcc", "gxx") or package:has_tool(toolkind, "gfortran")) then
+                    if package:has_tool(toolkind, "gcc", "gxx") then
+                        var_ldflags = 'ldflags'
```

Comment:
use `""` instead of `''`

Reply:
and please define it as local variable.

`local var_ldflags`

Reply:
I should have applied all the corrections.  I had to change slightly the layout of the code to have the variables available in the parent block.